comp/core/tagger/types/entity_id.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package types defines types used by the Tagger component.
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	taggerutils "github.com/DataDog/datadog-agent/pkg/util/tagger"
+)
+
+// EntityID represents a tagger entityID
+// An EntityID should be identified by a prefix and an id, and is represented as {prefix}://{id}
+type EntityID interface {
+	// GetID returns a prefix-specific id (i.e. an ID unique given prefix)
+	GetID() string
+	// GetPrefix returns the prefix of the EntityID
+	GetPrefix() EntityIDPrefix
+	// String returns a string representation of EntityID under the format {prefix}://{id}
+	String() string
+}
+
+// defaultEntityID implements EntityID as a plain string id
+type defaultEntityID string
+
+// GetID implements EntityID#GetID
+func (de defaultEntityID) GetID() string {
+	parts := strings.SplitN(string(de), "://", 2)
+
+	if len(parts) != 2 {
+		return ""
```

Comment:
Question: Would there be any side effect of returning and empty string for `GetID()` and ` GetPrefix()`?

Reply:
The condition `len(parts) != 2` will be `true` only when the entity id doesn't respect the correct format of the tagger defined entity ids.

The acceptable id's are defined [here](https://github.com/DataDog/datadog-agent/tree/main/comp/core/tagger#entity-ids). The id should be of the format `{prefix}://{id}`. `prefix` and `id` should not include the `://` separator. 

If for some reason we declare an entity id that doesn't respect the format, and an empty prefix (or id) is returned, the side-effect would be not being able to store/fetch tags with this entity id, which is totally fine since it is not supposed to work for malformatted entity ids

Reply:
> If for some reason we declare an entity id that doesn't respect the format, and an empty prefix (or id) is returned, the side-effect would be not being able to store/fetch tags with this entity id, which is totally fine since it is not supposed to work for malformatted entity ids

Sounds good. Do we have logging or something that would alert of that case?

Reply:
No problem. 

There was no logging in the tagger previously when an external caller calls the tagger with a malformatted entity id. 

In this situation, what was happening is that no tags are returned by the tagger, and the caller will (possibly) log a warning or error (depending on the situation) saying that tags were not found for this entity id.

In any case, the plan is to deprecate the plain string implementation of entity id, and to always use the composite implementation, which will not suffer from malformatted entity id's because the prefix and id will be stored in separate fields, no need to split or join strings.

Reply:
The composite entity id is defined [here](https://github.com/DataDog/datadog-agent/blob/ba27a220b174de70d63d38302067f1c9b70f3358/comp/core/tagger/types/entity_id.go#L61-L65)